File: JayKay24/Sorting 1/assignment-1.ts
import {
  Comparator,
  LessThan,
  GreaterThan,
} from "./comparator";

/**
 * Determine if a collection is sorted
 *
 * @param integers - an array of integers
 * @param comparator - order criteria, descending or ascending
 *
 * @returns Boolean indicating if the array is sorted
 */
function isSorted(
  integers: number[],
  comparator: Comparator<number> = new GreaterThan()
): boolean {
  for (let i = 1; i < integers.length; i++) {
    if (!comparator.compare(integers, i, i - 1)) return false;
  }

  return true;
}
/**
 * Time - O(n)
 * Space - O(1)
 */

// is an empty list, by definition, sorted?

const gt = new GreaterThan();
const lt = new LessThan();
console.log(isSorted([3, 2, 1], gt)); // false
console.log(isSorted([3, 2, 1], lt)); // true
console.log(isSorted([1, 4, 2, 3], gt)); // false

File: Shizler/Sorting 1/assignment_1.py
# implement a function to check a collection is sorted
# checking if array is sorted in ascending order

def isSorted(arr):
    n = len(arr)

    if n == 0 or n == 1:
        return True
    
    for i in range(len(arr)-1):
        if arr[i] > arr[i+1]:
            return False
    return True

print(isSorted([1,2,2,4,3]) )